sktime/transformations/series/detrend/_detrend.py

```diff
@@ -136,7 +152,11 @@ def _get_fitted_forecaster(self, X, y, fh):
         if self.forecaster_.get_tag("requires-fh-in-fit", True):
             X = update_data(self._X, X)
             y = update_data(self._y, y)
-            forecaster = self.forecaster_.clone().fit(y=X, X=y, fh=fh)
+            if not self.oos:
+                forecaster = self.forecaster_.clone().fit(y=X, X=y, fh=fh)
+            else:
+                forecaster = self.forecaster_.clone().fit(y=X[:10], X=y, fh=fh)
```

Comment:
hm, should this not be fitted to all the data, rather than just `X[:10]`?

Reply:
Because I think the whole point of this feature was to feed the data into forecaster in parts and get prediction in parts like the diagram I made in the [discussion](https://github.com/sktime/sktime/issues/4054#issuecomment-1426856960) we had earlier
![image](https://github.com/sktime/sktime/assets/86467990/ff71c427-3de6-4ef1-a38c-4f98705fa8a3)

Reply:
I see - but wouldn't we want to do it then only in `transform`, rather than in `fit`?

Reply:
I am not sure what you mean by this, kindly elaborate, please.

Reply:
I see! I was looking at the use of this functionality in a different way, I got it now! I will make the necessary changes now.